### PR TITLE
Revamp Tags LeftPanel design similar to glossaries

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes :

-  Revamp tags left panel similar to glossaries
- created generic leftpanel component.

https://github.com/open-metadata/OpenMetadata/issues/7128

#
### Type of change :

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx [hunk 1]
```diff
@@ -447,44 +446,40 @@ const TagsPage = () => {
...
-              </Button>
-            </Tooltip>
+      <LeftPanelCard id="tags">
+        <div className="tw-py-2" data-testid="data-summary-container">
+          <div className="tw-px-3">
+            <div className="tw-flex tw-justify-between tw-items-center">
+              <h6 className="tw-heading tw-text-sm tw-font-semibold">
+                Tag Categories
+              </h6>
+            </div>
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx:456]: do we need div wrapping here?

File: openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx [hunk 2]
```diff
@@ -525,28 +520,25 @@ const TagsPage = () => {
...
                 className="tw-flex tw-justify-between tw-items-center"
                 data-testid="header">
                 <div
-                  className="tw-heading tw-text-link tw-text-base"
+                  className="tw-text-link tw-text-base tw-py-2"
                   data-testid="category-name">
                   {currentCategory.displayName ?? currentCategory.name}
                 </div>
-                <div>
+                <div className="tw-flex tw-justify-between tw-items-center">
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx:533]:
```suggestion
                <div className="flex-center">
```

**Review by ShaileshParmar11 — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx (hunk 1 above)
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx]: Updated

File: openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx (hunk 2 above)
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/pages/tags/index.tsx]: Updated

**Pull request merged**